crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -2287,6 +2293,38 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn does_not_duplicate_global_registration_2311() {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, _http_proxy) = create_c8y_converter(&tmp_dir).await;
+
+        let measurement_message = Message::new(
+            &Topic::new_unchecked("te/device/main/service/my_measurement_service/m/my_type"),
+            r#"{"temperature": 21.37}"#,
+        );
+
+        // when auto registrating, local and cloud registration messages should be produced
+        let mapped_messages = converter.convert(&measurement_message).await;
+
+        let local_registration_message = mapped_messages
+            .iter()
+            .find(|m| EntityRegistrationMessage::new(m).is_some())
+            .unwrap();
+
+        // check if cloud registration message
+        assert!(mapped_messages
+            .iter()
+            .any(|m| m.topic.name == "c8y/s/us" && m.payload_str().unwrap().starts_with("102")));
+
+        // when converting a registration message the same as the previous one, no additional registration messages should be produced
+        let mapped_messages = converter.convert(local_registration_message).await;
+
+        let second_registration_message_mapped = mapped_messages.into_iter().any(|m| {
+            m.topic.name.starts_with("c8y/s/us") && m.payload_str().unwrap().starts_with("102")
+        });
+        assert!(!second_registration_message_mapped);
```

Comment:
```suggestion
        assert!(mapped_messages.is_empty());
```
That iteration over `mapped_messages` gives the feeling that some other mapped messages could still be there in it, whereas it shouldn't have any.

Reply:
I found that often times when we add new features that result in new messages being published, tests which assume the number of messages published need to get updated. Here I just wanted to assert that the registration message for this particular service was not produced again, with as few additional assumptions as possible.